pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -94,18 +99,57 @@ func (a *AWSActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *hi
 		computePool.Platform.AWS.Zones = zones
 	}
 
-	// Private subnets && userTags are settings available in the installconfig
-	// that we are choosing to ignore for the timebeing. These empty settings
-	// should be updated to feed from the machinepool / installconfig in the
-	// future.
 	subnets := map[string]string{}
+	// Fetching private subnets from the machinepool and then mapping availability zones to subnets
+	if len(pool.Spec.Platform.AWS.Subnets) > 0 {
+		subnetsByAvailabilityZone, err := a.getSubnetsByAvailabilityZone(pool)
+		if err != nil {
+			return nil, false, errors.Wrap(err, "describing subnets")
+		}
+		subnets = subnetsByAvailabilityZone
+	}
+	// userTags are settings available in the installconfig that we are choosing
+	// to ignore for the timebeing. These empty settings should be updated to feed
+	// from the machinepool / installconfig in the future.
 	userTags := map[string]string{}
 
 	installerMachineSets, err := installaws.MachineSets(cd.Spec.ClusterMetadata.InfraID, cd.Spec.Platform.AWS.Region, subnets, computePool, pool.Spec.Name, workerUserData, userTags)
 	if err != nil {
+		if strings.Contains(err.Error(), "no subnet for zone") {
+			conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+				pool.Status.Conditions,
+				hivev1.InvalidSubnetsMachinePoolCondition,
+				corev1.ConditionTrue,
+				"NoSubnetForAvailabilityZone",
+				err.Error(),
+				controllerutils.UpdateConditionIfReasonOrMessageChange,
+			)
+			if changed {
+				pool.Status.Conditions = conds
+				if err := a.client.Status().Update(context.Background(), pool); err != nil {
```

Comment:
Updating the MachinePool in the actuator is going to cause conflicts in the controller if the controller then tries to update the MachinePool.

Reply:
I am not able to understand the reasoning behind it. I believe the Machinepool object is passed as a reference to the actuator. We are updating the Machinepool in the GCP actuator as well. Will that also need to change?

Reply:
You are right. Disregard. :)